database/rrdhost.c

```diff
@@ -569,6 +569,10 @@ void rrdhost_system_info_free(struct rrdhost_system_info *system_info) {
         freez(system_info->host_os_version);
         freez(system_info->host_os_version_id);
         freez(system_info->host_os_detection);
+        freez(system_info->host_cores);
+        freez(system_info->host_cpu_freq);
+        freez(system_info->host_ram_total);
+        freez(system_info->host_disk_space);
```

Comment:
I don't like how `rrdhost_set_system_info_variable()` always frees whatever is in every `system_info` field, relying on the undocumented fact that if something hasn't been set yet it should be NULL.